pkg/webhook/cm/plugins/sloconfig/resource_qos_checker.go

```diff
@@ -111,6 +115,254 @@ func CheckNetQosCfg(cfg *configuration.ResourceQOSCfg, fldPath *field.Path) erro
+			}
+			if cfg.ClusterStrategy.LSClass.NetworkQOS.EgressRequest != nil && cfg.ClusterStrategy.LSClass.NetworkQOS.EgressRequest.Type == intstr.Int {
+				egressRequestTotal -= cfg.ClusterStrategy.LSClass.NetworkQOS.EgressRequest.IntValue()
+			}
+		}
+		if cfg.ClusterStrategy.BEClass != nil && cfg.ClusterStrategy.BEClass.NetworkQOS != nil {
+			if cfg.ClusterStrategy.BEClass.NetworkQOS.IngressRequest != nil && cfg.ClusterStrategy.BEClass.NetworkQOS.IngressRequest.Type == intstr.Int {
+				ingressRequestTotal -= cfg.ClusterStrategy.BEClass.NetworkQOS.IngressRequest.IntValue()
+			}
+			if cfg.ClusterStrategy.BEClass.NetworkQOS.EgressRequest != nil && cfg.ClusterStrategy.BEClass.NetworkQOS.EgressRequest.Type == intstr.Int {
+				egressRequestTotal -= cfg.ClusterStrategy.BEClass.NetworkQOS.EgressRequest.IntValue()
+			}
+		}
+
+		if ingressRequestTotal < 0 || egressRequestTotal < 0 {
+			return fmt.Errorf("ClusterStrategy total net bandwidth percentage already over 100")
+		}
+	}
+
+	// check netqos config is valid for each node strategy
+	for idx, nodeStrategy := range cfg.NodeStrategies {
+		ingressRequestTotal, egressRequestTotal := 100, 100
+		if nodeStrategy.LSRClass != nil && nodeStrategy.LSRClass.NetworkQOS != nil {
+			if nodeStrategy.LSRClass.NetworkQOS.IngressRequest != nil && nodeStrategy.LSRClass.NetworkQOS.IngressRequest.Type == intstr.Int {
+				ingressRequestTotal -= nodeStrategy.LSRClass.NetworkQOS.IngressRequest.IntValue()
+			}
+			if nodeStrategy.LSRClass.NetworkQOS.EgressRequest != nil && nodeStrategy.LSRClass.NetworkQOS.EgressRequest.Type == intstr.Int {
+				egressRequestTotal -= nodeStrategy.LSRClass.NetworkQOS.EgressRequest.IntValue()
+			}
+		}
+		if nodeStrategy.LSClass != nil && nodeStrategy.LSClass.NetworkQOS != nil {
+			if nodeStrategy.LSClass.NetworkQOS.IngressRequest != nil && nodeStrategy.LSClass.NetworkQOS.IngressRequest.Type == intstr.Int {
+				ingressRequestTotal -= nodeStrategy.LSClass.NetworkQOS.IngressRequest.IntValue()
+			}
+			if nodeStrategy.LSClass.NetworkQOS.EgressRequest != nil && nodeStrategy.LSClass.NetworkQOS.EgressRequest.Type == intstr.Int {
+				egressRequestTotal -= nodeStrategy.LSClass.NetworkQOS.EgressRequest.IntValue()
+			}
+		}
+		if nodeStrategy.BEClass != nil && nodeStrategy.BEClass.NetworkQOS != nil {
+			if nodeStrategy.BEClass.NetworkQOS.IngressRequest != nil && nodeStrategy.BEClass.NetworkQOS.IngressRequest.Type == intstr.Int {
+				ingressRequestTotal -= nodeStrategy.BEClass.NetworkQOS.IngressRequest.IntValue()
+			}
+			if nodeStrategy.BEClass.NetworkQOS.EgressRequest != nil && nodeStrategy.BEClass.NetworkQOS.EgressRequest.Type == intstr.Int {
+				egressRequestTotal -= nodeStrategy.BEClass.NetworkQOS.EgressRequest.IntValue()
+			}
+		}
+		if ingressRequestTotal < 0 || egressRequestTotal < 0 {
+			return fmt.Errorf("cfg.NodeStrategies[%d] total net bandwidth percentage already over 100", idx)
+		}
```

Comment:
It looks the same as the code in line 137-162. Should you consider extracting it into an independent function?

Reply:
done